### PR TITLE
prepare for 0.4.14 release

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
@Stebalien wanna do a quick look over this and see if anything is missing?

**Comment by Stebalien** [comment]
Should merge https://github.com/ipfs/go-ipfs/pull/4725 first.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,111 @@
 # go-ipfs changelog
 
+## 0.4.14 2018-03-02
+
+TODO: writeup
+
+- Features
+  - Pubsub IPNS Publisher and Resolver ([ipfs/go-ipfs#4047](https://github.com/ipfs/go-ipfs/pull/4047))
```
Comment (Stebalien) [inline CHANGELOG.md]: (experimental)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,111 @@
...
+  - Add uuid to event logs ([ipfs/go-ipfs#4392](https://github.com/ipfs/go-ipfs/pull/4392))
+  - Add --quiet flag to object put ([ipfs/go-ipfs#4411](https://github.com/ipfs/go-ipfs/pull/4411))
+  - Pinning memory improvements and fixes ([ipfs/go-ipfs#4451](https://github.com/ipfs/go-ipfs/pull/4451))
+  - Update WebUI version ([ipfs/go-ipfs#4449](https://github.com/ipfs/go-ipfs/pull/4449))
+  - Check strong and weak ETag validator ([ipfs/go-ipfs#3983](https://github.com/ipfs/go-ipfs/pull/3983))
+  - Improve and refactor FD limit handling ([ipfs/go-ipfs#3801](https://github.com/ipfs/go-ipfs/pull/3801))
+  - Support linking to non-dagpb objects in ipfs object patch ([ipfs/go-ipfs#4460](https://github.com/ipfs/go-ipfs/pull/4460))
+  - Improve allocation patterns of slices in bitswap ([ipfs/go-ipfs#4458](https://github.com/ipfs/go-ipfs/pull/4458))
+  - Secio handshake now happens synchronously ([libp2p/go-libp2p-secio#25](https://github.com/libp2p/go-libp2p-secio/pull/25))
+  - Don't hold locks while closing msgio pool ([libp2p/go-msgio#7](https://github.com/libp2p/go-msgio/pull/7))
```
Comment (Stebalien) [inline CHANGELOG.md]: "Don't block connection closing on pending writes" would more accurately reflect the impact of this change.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,111 @@
...
+  - Add godocs for routing module ([ipfs/go-ipfs#4676](https://github.com/ipfs/go-ipfs/pull/4676))
+  - Add godocs for path module ([ipfs/go-ipfs#4689](https://github.com/ipfs/go-ipfs/pull/4689))
+  - Add godocs for pin module ([ipfs/go-ipfs#4696](https://github.com/ipfs/go-ipfs/pull/4696))
+  - Update link to filestore experimental status ([ipfs/go-ipfs#4557](https://github.com/ipfs/go-ipfs/pull/4557))
+- Bugfixes
+  - Remove trailing slash in ipfs get paths, fixes #3729 ([ipfs/go-ipfs#4365](https://github.com/ipfs/go-ipfs/pull/4365))
+  - fix deadlock in bitswap sessions ([ipfs/go-ipfs#4407](https://github.com/ipfs/go-ipfs/pull/4407))
+  - Fix two race conditions (and possibly go routine leaks) in commands ([ipfs/go-ipfs#4406](https://github.com/ipfs/go-ipfs/pull/4406))
+  - Fix output delay in ipfs pubsub sub ([ipfs/go-ipfs#4402](https://github.com/ipfs/go-ipfs/pull/4402))
+  - AddWithContext wraps custom context ([ipfs/go-ipfs#4433](https://github.com/ipfs/go-ipfs/pull/4433))
```
Comment (Stebalien) [inline CHANGELOG.md]: "Use correct context in AddWithContext"

**Comment by whyrusleeping** [comment]
random test failures are random?

**Comment by whyrusleeping** [comment]
Yeah, heres the circle CI failure:
```
expecting success: 
    random-files -depth=4 -dirs=5 -files=8 foobar > /dev/null
  
error: mkdir foobar/49bukbth1-ey4/19jnx10c7xu3gx6/ptz7: not a directory
not ok 46 - create a bunch of random files
#	
#	    random-files -depth=4 -dirs=5 -files=8 foobar > /dev/null
#	  
```

It looks like there was a collision in the randomly generated named.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,162 @@
...
+round of the refactor disentangled the commands code from core ipfs code,
+allowing us to move it out into a [separate
+repository](https://github.com/ipfs/go-ipfs-cmds).  The code was previously
+very entangled with the go-ipfs codebase and not usable for other projects.
+The second round of the refactor had the goal of fixing several major issues
+around streaming outputs, progress bars, and error handling. It also paved the
+way for us to more easily provide an API over other transports, such as
+websockets and unix domain sockets.  It took a while to flush out all the kinks
+on such a massive change.  We're pretty sure we've got most of them, but if you
+notice anything weird, please let us know.
```
Comment (Stebalien) [inline CHANGELOG.md]: "most"

**Pull request merged**